modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/SeqexecEngine.scala

```diff
@@ -92,35 +90,51 @@ class SeqexecEngine(settings: SeqexecEngine.Settings) {
     type SequenceAR = engine.Sequence[engine.Action \/ engine.Result]
     type StepAR = engine.Step[engine.Action \/ engine.Result]
 
+    def viewSequence(seq: SequenceAR, st: SequenceState): SequenceView = {
+
+      def engineSteps(seq: SequenceAR): List[Step] = {
+
+        def viewStep(step: StepAR): Step =
+          StandardStep(
+            step.config,
+            engine.Step.status(step),
+            // TODO: Implement breakpoints at Engine level
+            breakpoint = false,
+            // TODO: Implement skipping at Engine level
+            skip = false,
+            configStatus = Map.empty,
+            // TODO: Implement standard step at Engine level
+            observeStatus = ActionStatus.Pending
+          )
 
-    def viewSequence(seq: SequenceAR, st: SequenceState): SequenceView =
+        // Couldn't find this on Scalaz
+        def splitWhere[A](l: List[A])(p: (A => Boolean)): (List[A], List[A]) =
+          l.splitAt(l.indexWhere(p))
+
+        seq.steps.map(viewStep) match {
```

Comment:
Yes, the engine doesn't need, so far, the concept of Step status. The status is only set at the top level, now the Sequence, and flows downstream from there. Aside of not having to keep coordinating separate statuses, it's also impossible to make them inconsistent (i.e. A Sequence Pending but with a Step Running).